### PR TITLE
Optimize & cleanup

### PR DESCRIPTION
**Pull request by siddharthteli12**
One more pr in reference to #421

### CODE REVIEW

File: pallets/automation-time/src/benchmarking.rs [hunk 1]
```diff
@@ -306,8 +306,8 @@ benchmarks! {
 		let collator: T::AccountId = account("collator", 0, SEED);
 		let account_minimum = T::Currency::minimum_balance().saturating_mul(ED_MULTIPLIER.into());
 
-		T::Currency::deposit_creating(&delegator, (10_000_000_000_000_000 as u128).saturated_into());
-		T::Currency::deposit_creating(&collator, (100_000_000_000_000_000 as u128).saturated_into());
+		T::Currency::deposit_creating(&delegator, (10_000_000_000_000_000_u128).saturated_into());
```
[the remark this answers is not in the document]
Comment (siddharthteli12) [reply pallets/automation-time/src/benchmarking.rs:309]: The value is used only once. Declaring it as a const would just increase the overhead in my opinion.

File: pallets/automation-time/src/lib.rs [hunk 2]
```diff
@@ -1314,25 +1314,22 @@ pub mod pallet {
...
-					// Only native token are supported as the XCMP fee for local deductions
-					if instruction_sequence == InstructionSequence::PayThroughSovereignAccount &&
-						asset_location != MultiLocation::new(0, Here)
-					{
-						Err(Error::<T>::UnsupportedFeePayment)?
-					}
-				},
-				_ => (),
-			};
+			if let Action::XCMP { execution_fee, instruction_sequence, .. } = action.clone() {
```
[the remark this answers is not in the document]
Comment (siddharthteli12) [reply pallets/automation-time/src/lib.rs:1317]: Looks like it's not possible, at least without functional changes I believe.

File: pallets/automation-time/src/types.rs [hunk 3]
```diff
@@ -249,8 +249,6 @@ impl<AccountId: Clone, Balance> Task<AccountId, Balance> {
 		instruction_sequence: InstructionSequence,
 		abort_errors: Vec<Vec<u8>>,
 	) -> Result<Self, DispatchError> {
-		let destination =
-			MultiLocation::try_from(destination).map_err(|_| Error::<T>::BadVersion)?;
```
[the remark this answers is not in the document]
Comment (siddharthteli12) [reply pallets/automation-time/src/types.rs:253]: @v9n, Not in this case I believe. But yeah, it was necessary when destination's type was `VersionedMultiLocation`. For eg - https://github.com/OAK-Foundation/OAK-blockchain/blob/cf4a6eadc6a12fa02153434f71d34bf3e6bf9ac0/pallets/automation-time/src/lib.rs#L387-L388

**Comment by siddharthteli12** [comment]
@v9n @imstar15 , Its ready for review. I believe we can merge this pr now as it's been pending for a while now.

**Comment by siddharthteli12** [comment]
@v9n , Resolved the conflict.

**Comment by chrisli30** [comment]
Thank you both. The PR is sync’ed with main so I’m merging it.

**Pull request merged**